w4_d1_1_intro_to_mongo/README.md

```diff
@@ -0,0 +1,169 @@
+# Intro to Mongo
+
+| Objectives |
+| :--- |
+| Install and use MongoDB with a node/express app |
+| Describe and explain what an ORM is and why it rocks. |
+| Perform Read & Write operations on a model |
+
+| Concepts | Tools | Activities |
+| :---: | :---: | :---: |
+| Document-based Database, Schemaless Database, Models | MongoDB, Mongoose | Challenges |
+
+### Motivation (Why?)
+
+NoSQL databases are a popular and trending database technology and it gives us a full Javascript/JSON stack.
+
+* Node - JavaScript
+* Express - JavaScript
+* jQuery - JavaScript
+* AJAX - JSON (JavaScript)
+* **Mongo - JSON (JavaScript)**
+* AngularJS - JavaScript
+
+### Analogy (What?)
+
+Imagine you're going to make a cake.
+
+SQL is like shopping for the ingredients for the recipe. At the store, you have to go to many different aisles to get each ingredient.
+
+Wouldn’t it be nicer if there was a store was organized by recipe, so you could go to one place in the store and grab everything you need from that one spot? That would be nice. And that would be like a NoSQL database.
+
+[mgoffin/Stack Overflow](http://stackoverflow.com/questions/14428069/sql-and-nosql-analogy-for-the-non-technical/14428221#14428221)
+
+> NOTE: a "model" is like a JavaScript constructor for objects we will save to our database. Models are the 'data layer' of MVC frameworks.
+
+### Key Snippets
+
+##### Enter a Node Server Console (REPL Console - Read-Eval-Print-Loop Console)
+```
+$ node
+```
+
+##### Connect a Server to Your Database
+```
+//
+// index.js or server.js
+//
+
+var mongoose = require("mongoose");
+mongoose.connect("mongodb://localhost/test");
+```
+
+##### Make a Model
+```
+//
+//  models/book.js
+//
+
+var Schema = mongoose.Schema;
+var DinosaurSchema = new Schema({
+    name: String,
+    period: String,
+    description: String,
+    height_in_inches: Integer
+});
+
+mongoose.model('Dinosaur', DinosuarSchema);
```

Comment:
What does this do?  In the docs I always see it saved as a variable (e.g., Dinosaur = mongoose.model('Dinosaur',DinosaurSchema)).  

I think we should only call mongoose.model() once -- either save the result of mongoose.model() as a variable in dinosaur.js and use module.exports to export it, or use the line IN ANOTHER FILE (below) that calls mongoose.model(). 

My preference would be to use exports, so this line 67 becomes var Dinosaur = mongoose.model('Dinosaur', DinosaurSchema)  and the IN ANOTHER FILE line 70 becomes var Dinosaur = require('dinosaur.js').